### DISCUSSION TITLE
Init called twice

### DISCUSSION THREAD

**Discussion by PatrikFomin** [alpinejs/alpine#3474] [1. Help] [answered]
I am trying to figure out if I have written it wrong, or if it's something else. (There has been several topics about init being called twice but I haven't seen any good responses so far that matches.)

It's a simple menu. There is a x-data object on the div just before the menu starts and then there are a few html templates for some for loops and to display the content. Nothing advanced. Also, it works fine and outputs exactly what it should do, just trying to mitigate the init from being called twice.

I have highlighted all tags with Alpine commands.

Head
![html2](https://user-images.githubusercontent.com/18208521/226101751-b07cc20c-58c2-48a7-a0a1-940b197e9b2d.png)
Content
![html](https://user-images.githubusercontent.com/18208521/226101755-fe383ca4-2b41-4232-8072-a24f16beb041.png)
JS
(Init gets called twice)
![js](https://user-images.githubusercontent.com/18208521/226101757-83e89763-98f0-42f3-9517-583f440b0d6a.png)
Chrome debug
![chrome](https://user-images.githubusercontent.com/18208521/226101861-d8e3beb4-b03e-4ccd-b7c3-527bfeb89528.png)

**Comment by SimoTod** [comment]
Are you using the alpine extension for chrome? I think it's not reliable with v3. If you disinstall it (and put a debug console log in your init function), you should see that it's just called once

**Reply by PatrikFomin** [reply]
I wish it were. I'm using version 3 however I found it using the journal debugging of the JavaScript class in Chrome. The await fetch gets triggered twice in there, and the response "then" is executed twice.

**Comment by joshhanley** [comment]
@PatrikFomin any chance you could make a codepen or jsfiddle of the issue, with as little code as possible, to make it easier for others to have a look? Thanks!

**Reply by PatrikFomin** [reply]
Of course I can't reproduce it in jsfiddle. However I will try to put it live on the server and send the link.

**Reply by ekwoka** [reply]
It's almost definitely because Alpine is being started twice.

Only other explanation is something is removing and readding the element.

**Reply by PatrikFomin** [reply]
But what can cause Alpine to be started twice? (the script is only included once, and the html layout template we use doesn't use Alpine JS)
2x In here is from the Init in the menu file.
![image](https://user-images.githubusercontent.com/18208521/227163779-443b6f77-e0a0-4643-9749-3433491a2779.png)

**Reply by ekwoka** [reply]
Well, idk what is in that script there, or what's in anything else there.

But it's most likely that `Alpine.start` is being called twice. I believe it does init twice even if there aren't actually two instances of Alpine entirely.

**Reply by PatrikFomin** [reply]
I found that if I out-comment the plugins from the html theme we use, then it stops being called twice.
Somehow the plugins.js seems to bind or in some other way interfere with Alpine's start/init. Plugins doesn't contain Alpine, or has anything to do with Alpine but somehow it causes the init to trigger twice.

https://test-website.nordicfuzzcon.org/Test/Index

**Reply by SimoTod** [reply]
The generated HTML in the dev console shows 2 components with `x-data="alpineMenu"` so the function is called correctly once for every component. I suspect it's your plugin library that clones the dom node since it only show` once on the source code: `view-source:https://test-website.nordicfuzzcon.org/Test/Index`

**Reply by PatrikFomin** [reply]
You are correct. If I move the alpineMenu up to the header-tag then it just get's called once.

**Comment by PatrikFomin** [comment, marked as answer]
Resolved with the answer from @SimoTod above. 
A plugin that was used for the website clone the dom and this causes a second call to be triggered. Moving it up to another tag solved the issue.
